### PR TITLE
feat: kubespan node endpoints filter

### PR DESCRIPTION
**Pull request by sergelogvinov**
This feature allows us to use only IPv4 or IPv6 stack to reach the peers. Also, it can help to not share the node-specific IPs, which cannot be accessible at all.

part of #5937, #4898, #4369

I've renamed the key filterSubnets -> filters. Because I believe it will have more options inside... 

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: pkg/machinery/config/types/v1alpha1/v1alpha1_types.go [hunk 1]
```diff
@@ -2505,6 +2505,15 @@ type NetworkKubeSpan struct {
...
 	KubeSpanMTU *uint32 `yaml:"mtu,omitempty"`
+	// description: |
+	//   KubeSpan filters.
+	KubeSpanFilters *KubeSpanFilters `yaml:"filters,omitempty"`
+}
+
+// KubeSpanFilters struct describes KubeSpan filters.
+type KubeSpanFilters struct {
+	// description: KubeSpanFiltersEndpoints list allowed node' IPs to make p2p connections.
+	KubeSpanFiltersEndpoints *[]string `yaml:"endpoints,omitempty"`
```
Comment (smira) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_types.go:2516]: nit: as it stands out, you don't need a pointer to slice, as slice itself will be `nil` if it's not specified.

**Comment by sergelogvinov** [comment]
I've added tests and example to use.

**Review by smira — APPROVED** [review]

**Pull request merged**